firebase-firestore/src/main/java/com/google/firebase/firestore/bundle/BundleReader.java

```diff
@@ -133,9 +134,9 @@ private BundleElement readNextElement() throws IOException, JSONException {
       throw abort("Reached the end of bundle when a length string is expected.");
     }
 
-    char[] c = new char[nextOpenBracket];
+    byte[] c = new byte[nextOpenBracket];
     buffer.get(c);
-    return new String(c);
+    return charset.decode(ByteBuffer.wrap(c)).toString();
```

Comment:
Note that `indexOfOpenBracket` returns the char-index of the first bracket. Using a byte buffer here is not correct. but likely won't make a difference since this part of the input should be ASCII.

Reply:
See above.